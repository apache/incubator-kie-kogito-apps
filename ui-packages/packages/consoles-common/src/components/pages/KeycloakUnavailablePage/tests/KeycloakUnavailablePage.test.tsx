import { Button } from '@patternfly/react-core';
import { mount } from 'enzyme';
import React from 'react';
import KeyCloakUnavailablePage from '../KeycloakUnavailablePage';

describe('KeycloakUnavailablePage test', () => {
  it('render the page', () => {
    const wrapper = mount(<KeyCloakUnavailablePage />);
    expect(wrapper).toMatchSnapshot();
  });
  it('reload button is clicked', () => {
    const location: Location = window.location;
    delete window.location;
    window.location = {
      ...location,
      reload: jest.fn()
    };
    const wrapper = mount(<KeyCloakUnavailablePage />);
    wrapper
      .find(Button)
      .find('button')
      .simulate('click');
    expect(window.location.reload).toHaveBeenCalled();
    jest.restoreAllMocks();
    window.location = location;
  });
});
