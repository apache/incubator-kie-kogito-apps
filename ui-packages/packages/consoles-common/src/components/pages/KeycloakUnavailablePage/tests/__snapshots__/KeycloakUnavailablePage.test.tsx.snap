// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`KeycloakUnavailablePage test render the page 1`] = `
<KeycloakUnavailablePage>
  <div
    data-ouia-component-type="server-unavailable"
    data-ouia-safe={true}
    style={
      Object {
        "background": "#868686",
        "display": "flex",
        "height": "100%",
        "justifyContent": "center",
        "paddingTop": "5%",
      }
    }
  >
     
    <Card
      style={
        Object {
          "maxHeight": "300px",
        }
      }
    >
      <article
        className="pf-c-card"
        data-ouia-component-id="OUIA-Generated-Card-1"
        data-ouia-component-type="PF4/Card"
        data-ouia-safe={true}
        id=""
        style={
          Object {
            "maxHeight": "300px",
          }
        }
      >
        <CardHeader>
          <div
            className="pf-c-card__header"
          >
            <CardHeaderMain>
              <div
                className=""
              >
                <Brand
                  alt="Kogito keycloak"
                  src="kogito.png"
                  style={
                    Object {
                      "height": "30px",
                    }
                  }
                >
                  <img
                    alt="Kogito keycloak"
                    className="pf-c-brand"
                    src="kogito.png"
                    style={
                      Object {
                        "height": "30px",
                      }
                    }
                  />
                </Brand>
              </div>
            </CardHeaderMain>
          </div>
        </CardHeader>
        <CardBody
          isFilled={false}
        >
          <div
            className="pf-c-card__body pf-m-no-fill"
          />
        </CardBody>
        <CardTitle>
          <div
            className="pf-c-card__title"
          >
            Error:503 - Server unavailable
          </div>
        </CardTitle>
        <CardBody
          isFilled={false}
        >
          <div
            className="pf-c-card__body pf-m-no-fill"
          />
        </CardBody>
        <CardBody>
          <div
            className="pf-c-card__body"
          >
            Sorry.. the keycloak server seems to be down
          </div>
        </CardBody>
        <CardBody
          isFilled={false}
        >
          <div
            className="pf-c-card__body pf-m-no-fill"
          />
        </CardBody>
        <Bullseye>
          <div
            className="pf-l-bullseye"
          >
            <span>
              Please contact administrator or
               
              <Button
                isInline={true}
                onClick={[Function]}
                variant="link"
              >
                <button
                  aria-disabled={false}
                  aria-label={null}
                  className="pf-c-button pf-m-link pf-m-inline"
                  data-ouia-component-id="OUIA-Generated-Button-link-1"
                  data-ouia-component-type="PF4/Button"
                  data-ouia-safe={true}
                  disabled={false}
                  onClick={[Function]}
                  role={null}
                  type="button"
                >
                  click here to retry
                </button>
              </Button>
            </span>
          </div>
        </Bullseye>
        <CardBody
          isFilled={false}
        >
          <div
            className="pf-c-card__body pf-m-no-fill"
          />
        </CardBody>
      </article>
    </Card>
  </div>
</KeycloakUnavailablePage>
`;
